CMakePresets.json

```diff
@@ -0,0 +1,173 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "base",
```

Comment:
maybe `default` is a better name. Does this also decide the directory in `build/`, e.g. `build/default/` ?

Reply:
I have actually configured "base" as hidden, but I can switch it to "default" and make it visible. In that case, should we drop "debug", or what do you think is a good split? 

> Does this also decide the directory in `build/`, e.g. `build/default/` ?

yes, it uses [macro expansion](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#macro-expansion) to determine the build folder, but each preset can still have its own `binaryDir`

> An optional string representing the path to the output binary directory\. This field supports macro expansion\. If a relative path is specified, it is calculated relative to the source directory\. If binaryDir is not specified, it must be inherited from the inherits preset \(unless this preset is hidden\)\.

basically pick the layout you see fit for the build folder, and then we can add the rules to accommodate it.

Reply:
no need to change--it's good as you have it